ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/XATerminatorImple.java

```diff
@@ -94,23 +94,32 @@ public void commit (Xid xid, boolean onePhase) throws XAException
             SubordinateTransaction tx = SubordinationManager
                     .getTransactionImporter().getImportedTransaction(xid);
 
-            if (tx == null)
-                throw new XAException(XAException.XAER_INVAL);
+            if (tx == null) {
+                XAException xaException = new XAException("No subordinate transaction to  drive for commit with xid: " + xid);
```

Comment:
I think we should localise the exception messages

Reply:
ah, thanks! I haven't realized that.